packages/steiger/src/shared/globs/glob-matchers.ts

```diff
@@ -40,3 +40,14 @@ export function createFilterAccordingToGlobs({ inclusions, exclusions }: ApplyGl
 
   return filterAccordingToGlobs
 }
+
+/**
+ * Checks if a file/folder is a well-known ignore.
+ *
+ * If a path does not match the "any file/folder" pattern, it is considered a well-known ignore.
+ * It looks like minimatch has some registry of such files, and it returns false for them regardless of the pattern.
```

Comment:
note: I think what's happening here is that hidden files (those whose name starts with a dot) are not matched by globs by default. That's how terminal shells handle globs too. 

issue: I don't think ignoring hidden files it's the right approach here, since there are legitimate reasons for why you might want to make a source code file hidden.

suggestion: I'd rather investigate why these files are picked up at all. I was under the impression that we automatically disregard the files that are in .gitignore. If we don't, then we should, I think.

Reply:
Oh, yeah, it was so obvious to me that it should have ignored only default hidden FS-specific files that I didn't even  checked how it behaves for dot files as such. Yes, it turns out this happens with any dot files, not just a few specific ones. Given these new circumstances, yes, that's not the right approach at all.

Yes, I'll investigate why the files that are in .gitignore are not disregarded by default